PhysicsTools/PatUtils/python/tools/runMETCorrectionsAndUncertainties.py

```diff
@@ -736,19 +741,19 @@ def createEnergyScaleShiftedUpModule(self, process,identifier, objectCollection,
                     cms.PSet(
                         binSelection = cms.string('pdgId==130'),
                         energyDependency = cms.bool(True),
-                        binUncertainty = cms.string('((abs(y)<1.3)?(1/sqrt(x)+0.07):(1.5/sqrt(x)+0.04))')
+                        binUncertainty = cms.string('((abs(y)<1.3)?(max(0.25,sqrt(pow(0.8/sqrt(x), 2)+0.05*0.05))):(max(0.30,sqrt(pow(1.0/sqrt(x),2)+0.04*0.04))))')
```

Comment:
The full forumla has been written to reflect the usual calorimetric resolution formula made with a quadratic sum of the stochastic term and a constant term.

The edm producer code already contains the protection against events with x=0